build.rb

```diff
@@ -10,21 +10,44 @@
   'lang:fr_CA' => 'fr',
 }.freeze
 
-query = <<~SPARQL
+legislative_dir = Pathname.new(__FILE__).dirname.join('legislative')
+index_file = legislative_dir.join('index.json')
+
+def date_condition(start_date, end_date)
+  return '' unless start_date && end_date
```

Comment:
This seems like it could cause quite subtle problem if someone ever passes a`start_date` and no `end_date`. At the moment that's not a big issue (not least because the only place that calls it does its own validation of that), but it's generally best to not rely on external checks.

I don't think it's necessarily worth doing much with right now, but it might be worth leaving a TODO comment so that when we come to tidy this all up later, we don't overlook it.

Reply:
Actually, originally the implementation coped with `end_date` being `nil`, but I took that out for consistency with the calling code. I'll just put that back :)